packages/arb-avm-cpp/avm_values/include/avm_values/buffer.hpp

```diff
@@ -210,25 +210,31 @@ class RawBuffer {
 class Buffer {
    public:
     std::shared_ptr<RawBuffer> buf;
+    uint64_t maxAccess;
```

Comment:
Should this be the maximum index accessed, or the maximum index written?   I think written makes more sense, because we don't need to build the tree for reads beyond the end of the tree.

Reply:
Yeah, best change the name.